core/embed/sec/optiga/optiga.c

```diff
@@ -934,24 +1038,58 @@ optiga_pin_result optiga_pin_verify(
     goto end;
   }
 
-  // Reset the counter which limits the guesses at OID_STRETCHED_PIN.
-  if (optiga_set_data_object(OID_STRETCHED_PIN_CTR, false, COUNTER_RESET,
-                             sizeof(COUNTER_RESET)) != OPTIGA_SUCCESS) {
-    ret = OPTIGA_PIN_ERROR;
-    goto end;
+  // Reset the counter which limits the guesses at OID_STRETCHED_PINS.
+  if (pin_index == 0) {
+    if (optiga_reset_counter(OID_STRETCHED_PIN_CTR, PIN_MAX_TRIES) !=
+        OPTIGA_SUCCESS) {
+      ret = OPTIGA_PIN_ERROR;
+      goto end;
+    }
+  } else {
+    // The one extra attempt is used to authorize using the first stretched PIN
+    // in optiga_pin_reset_hmac_counter().
+    if (optiga_reset_counter(OID_STRETCHED_PIN_CTR, PIN_MAX_TRIES + 1) !=
+        OPTIGA_SUCCESS) {
+      ret = OPTIGA_PIN_ERROR;
+      goto end;
+    }
```

Comment:
See my other code improvement suggestion above.
```suggestion
  if (optiga_reset_counter(OID_STRETCHED_PIN_CTR, stretched_pin_ctr_limit) !=
      OPTIGA_SUCCESS) {
    ret = OPTIGA_PIN_ERROR;
    goto end;
```